python/plugins/db_manager/db_model.py

```diff
@@ -398,7 +398,9 @@ def flags(self, index):
         if index.column() == 0:
             item = index.internalPointer()
 
-            if isinstance(item, SchemaItem) or isinstance(item, TableItem):
+            if isinstance(item, SchemaItem) \
+                    or (isinstance(item, TableItem)
+                        and not (self.hasGPKGSupport and item.getItemData().type == Table.RasterType)):
```

Comment:
Not sure if this is the better way to check if the item is a raster layer in a GeoPackage container...

Reply:
DB manager is ancient code, a more modern approach would be to use the connections API:

```python
md = QgsProviderRegistry.instance().providerMetadata('postgres')
conn = md.createConnection("dbname='tinyows_demo' host=localhost port=5433 username='***' password='***'", {})
table = conn.table('public', 'demo3dlinestring')
bool(table.flags() & QgsAbstractDatabaseProviderConnection.TableFlag.Raster)
bool(table.flags() & QgsAbstractDatabaseProviderConnection.TableFlag.Vector)
```

It should work with all QGIS supported DB providers (GPKG included).